pkg/utils/net-attach-def.go

```diff
@@ -152,14 +152,41 @@ func CreateNetworkStatuses(r cnitypes.Result, networkName string, defaultNetwork
 	// Same for DNS
 	v1dns := convertDNS(result.DNS)
 
+	// Check for a gateway-associated interface, we'll use this later if we did to mark as the default.
+	gatewayInterfaceIndex := -1
+	if defaultNetwork {
+		for _, ipConfig := range result.IPs {
+			if ipConfig.Gateway != nil && ipConfig.Interface != nil {
+				// Keep the index of the first interface that has a gateway
+				gatewayInterfaceIndex = *ipConfig.Interface
+				break
+			}
+		}
+	}
+
 	// Initialize NetworkStatus for each container interface (e.g. with sandbox present)
 	indexOfFoundPodInterface := 0
 	foundFirstSandboxIface := false
+	didSetDefault := false
 	for i, iface := range result.Interfaces {
 		if iface.Sandbox != "" {
+			isDefault := false // default to false by default
+
+			// If there's a gateway listed for this interface index found in the ips, we mark that interface as default
+			// notably, we use the first one we find.
+			if defaultNetwork && i == gatewayInterfaceIndex && !didSetDefault {
+				isDefault = true
+				didSetDefault = true
+			}
+
+			// Otherwise, if we didn't find it, we use the first sandbox interface.
+			if defaultNetwork && gatewayInterfaceIndex == -1 && !foundFirstSandboxIface {
```

Comment:
shouldn't we also enable the `foundFirstSandboxIface` flag ? 

I know we're doing it later on, in line 202, but I think it would be more readable if we did it in this code block.

Reply:
Yeah this got spread out huh, good eye, agreed.